src/wp-includes/global-styles-and-settings.php

```diff
@@ -285,13 +285,8 @@ function wp_theme_has_theme_json() {
 	}
 
 	// Does the theme have its own theme.json?
-	$theme_has_support = is_readable( get_stylesheet_directory() . '/theme.json' );
-
-	// Look up the parent if the child does not have a theme.json.
-	if ( ! $theme_has_support ) {
-		$theme_has_support = is_readable( get_template_directory() . '/theme.json' );
-	}
-
+	$theme_has_support = is_readable( get_theme_file_path( 'theme.json' ) );
```

Comment:
See my overarching feedback: If we go with this approach, I think we should use `file_exists` throughout.
```suggestion
	$theme_has_support = file_exists( get_theme_file_path( 'theme.json' ) );
```